src/domain_bridge/wait_for_graph_events.hpp

```diff
@@ -91,100 +119,57 @@ class WaitForQosHandler
    *   If QoS settings are already available, then the callback is called immediately.
    */
   void register_on_publisher_qos_ready_callback(
-    std::string topic,
-    rclcpp::Node::SharedPtr node,
+    const std::string & topic,
+    const rclcpp::Node::SharedPtr & node,
```

Comment:
Just curious, why switch to a node (versus a shared pointer)? I suppose we don't have to worry about lifetime of the node in either case.

Reply:
This is a `Node::SharedPtr &`, not just a `Node &`.
Currently the `waiting_threads_` map is retaining part ownership of the `Node` if there wasn't a thread for that node running already.
The recommendation is to pass ` const shared_ptr<T> &` when you might retain ownership, so it matches the pattern.

---

Currently `WaitForGraphEvents` is retaining part ownership of the passed nodes.
It could also only borrow a reference to a node, and in that case external code should ensure that the node lives for long enough.
But that might make the code harder to understand, so the current approach seems fine.

Reply:
Makes sense, thanks for the explanation!